homeassistant/components/homekit_controller/connection.py

```diff
@@ -513,6 +514,54 @@ def async_remove_legacy_device_serial_numbers(self) -> None:
 
             device_registry.async_update_device(device.id, new_identifiers=identifiers)
 
+    @callback
+    def async_reap_stale_entity_registry_entries(self) -> None:
+        """Delete entity registry entities for removed characteristics, services and accessories."""
+        _LOGGER.debug(
+            "Removing stale entity registry entries for pairing %s",
+            self.unique_id,
+        )
+
+        reg = er.async_get(self.hass)
+
+        # For the current config entry only, visit all registry entity entries
+        # Build a set of (unique_id, aid, sid, iid)
+        # For services, (unique_id, aid, sid, None)
+        # For accessories, (unique_id, aid, None, None)
+        entries = er.async_entries_for_config_entry(reg, self.config_entry.entry_id)
+        exisiting_entities = {}
+        for entry in entries:
+            if iids := unique_id_to_iids(entry.unique_id):
+                exisiting_entities[iids] = entry.entity_id
+        existing_unique_id = set(exisiting_entities.keys())
```

Comment:
```suggestion
        existing_entities = {
            iids: entry.entity_id
            for entry in entries
            if (iids := unique_id_to_iids(entry.unique_id))
        }
```

Maybe write it as a dict comp?